components/FilterBar/index.tsx

```diff
@@ -16,20 +17,58 @@ export const FilterBar = ({
   filters: Filter[];
   onFilterChange: (filter: Filter) => void;
 }) => {
+  const [activeFilter, setActiveFilter] = useState<Filter | null>(null);
+
+  const [selectedFilters, setSelectedFilters] = useState<Filters>({
+    statusCompleted: false,
+    technology: [],
+    projectSize: { min: 0, max: 0 },
+    location: [],
+  });
+
+  const handleButtonClick = (filter: Filter) => {
+    setActiveFilter(activeFilter?.id === filter.id ? null : filter);
+    onFilterChange(filter);
+  };
+
+  useEffect(() => {
+    console.log(activeFilter);
+  }, [activeFilter]);
+
+  const handleTechnologyChange = (options: string[]) => {
+    setSelectedFilters(prevFilters => ({
+      ...prevFilters,
+      technology: options,
+    }));
+  };
   return (
-    <FilterBarStyles>
+    <FilterContainerStyles>
       {filters.map(filter => (
-        <div key={filter.id}>
-          <FilterButtonStyles
-            key={filter.label}
-            onClick={() => onFilterChange(filter)}
-          >
-            {filter.icon}
-            {filter.label}
-            <RiArrowDropDownLine />
-          </FilterButtonStyles>
-        </div>
+        <FilterBackgroundStyles
```

Comment:
can you turn this into a `Filter` component. Then you pass in the following props: label, isActive, icon, dropdown component (or what props you need to decide which filter it is). Right now the filter bar component looks messy. This will help further abstract and organize the different filters so that it's easier for people to make their own filter dropdowns.